azurelinuxagent/common/cgroupapi.py

```diff
@@ -557,40 +573,36 @@ def create_and_start_unit(unit_filename, unit_contents):
             raise CGroupsException("Failed to create and start {0}. Error: {1}".format(unit_filename, ustr(e)))
 
     @staticmethod
-    def _get_extensions_slice_root_name():
-        return "system-{0}.slice".format(EXTENSIONS_ROOT_CGROUP_NAME)
+    def _get_azure_slice_name():
+        return "{0}.slice".format(AZURE_CGROUP_NAME)
+
+    @staticmethod
+    def _get_extensions_root_slice_name():
+        return "{0}-{1}.slice".format(AZURE_CGROUP_NAME, EXTENSIONS_ROOT_CGROUP_NAME)
 
     def _get_extension_slice_name(self, extension_name):
-        return "system-{0}-{1}.slice".format(EXTENSIONS_ROOT_CGROUP_NAME, self._get_extension_cgroup_name(extension_name))
-
-    def create_extension_cgroups_root(self):
-        unit_contents = """
-[Unit]
-Description=Slice for walinuxagent extensions
-DefaultDependencies=no
-Before=slices.target
-Requires=system.slice
-After=system.slice"""
-        unit_filename = self._get_extensions_slice_root_name()
+        return "{0}-{1}-{2}.slice".format(AZURE_CGROUP_NAME, EXTENSIONS_ROOT_CGROUP_NAME, self._get_extension_cgroup_name(extension_name))
+
+    def create_azure_slice(self):
+        unit_contents = """[Unit]
+Description=Slice for Azure VM Agent and Extensions"""
+        unit_filename = self._get_azure_slice_name()
+
+        self.create_and_start_unit(unit_filename, unit_contents)
+        logger.info("Created root slice for Azure VM Agent and Extensions {0}".format(unit_filename))
+
+    def create_extensions_slice(self):
+        unit_contents = """[Unit]
+Description=Slice for Azure VM Extensions"""
+        unit_filename = self._get_extensions_root_slice_name()
+
         self.create_and_start_unit(unit_filename, unit_contents)
-        logger.info("Created slice for walinuxagent extensions {0}".format(unit_filename))
+        logger.info("Created root slice for Azure VM Extensions {0}".format(unit_filename))
```

Comment:
let's rename 'create_and_start_unit' to 'create_unit_file' and remove the calls to systemctl